<a href="https://colab.research.google.com/github/pedroyepiz/pythonHOY/blob/main/17_ARCHIVOS_DRIVE_Y_LOCALES.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!pip install pyyaml

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
from IPython.display import clear_output
import random
import json
import yaml
import time
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
ruta="/content/drive/MyDrive/proyecto"


In [ ]:
import sys
sys.path.append(ruta)


In [ ]:
import os

# Ruta de la carpeta "proyecto" en Google Drive
path = "/content/drive/MyDrive/proyecto"

# Nombre del archivo de texto a crear
filename = "peso_pluma.txt"

# Contenido del archivo
p1 = "Me levanto, un baño y luego me pongo a forjar\n"
p2="Es veneno, da pa' arriba, de muy buena calidad\n"
p3="Los teléfonos no paran nunca de sonar\n"
p4="Si no es alguna plebita, es porque un cliente quiere más\n"

contenido = p1 + p2 + p3 + p4

# Crear el archivo y escribir el contenido
with open(os.path.join(path, filename), "w") as file:
    file.write(contenido)


In [ ]:
import dicionario
listadatos= []
claves = ["Id","Ap.Paterno","Ap.Materno","Nombre","Puesto","Sexo"]
for i in range(15):
    listadatos.append(dict(zip(claves,dicionario.registro())))

In [ ]:
data = pd.DataFrame(listadatos)
print (data)


     Id Ap.Paterno  Ap.Materno   Nombre         Puesto    Sexo
0    71      Casas       Perez   Lupita        Gerente   MUJER
1   482    Sanchez     Sanchez     Luis      Encargad@  HOMBRE
2    48    Herrera     Sanchez  Abraham     Supervisor  HOMBRE
3   210    Herrera     Herrera    Maria  Empleado Gral   MUJER
4   266    Canales     Canales   Sandra       Conserje   MUJER
5   476      Perez     Canales  Abraham           Jefe  HOMBRE
6    73    Sanchez       Casas    Tulio        Guardia  HOMBRE
7   164    Paredes     Herrera    Sonia  Empleado Gral   MUJER
8    37      Casas   Gutierrez     Juan      Encargad@  HOMBRE
9   245    Herrera      Huerta      Ana        Gerente   MUJER
10   24  Gutierrez      Huerta    Sonia        Guardia   MUJER
11  440      Perez  Villalobos      Ana       Conserje   MUJER
12  155     Chavez     Paredes  Vanessa        Guardia   MUJER
13  143     Huerta     Sanchez   Carlos  Empleado Gral  HOMBRE
14  243      Casas     Herrera  Vanessa     Supervisor 

|   Id | Ap.Paterno   | Ap.Materno   | Nombre   | Puesto        | Sexo   |
|-----:|:-------------|:-------------|:---------|:--------------|:-------|
|   71 | Casas        | Perez        | Lupita   | Gerente       | MUJER  |
|  482 | Sanchez      | Sanchez      | Luis     | Encargad@     | HOMBRE |
|   48 | Herrera      | Sanchez      | Abraham  | Supervisor    | HOMBRE |
|  210 | Herrera      | Herrera      | Maria    | Empleado Gral | MUJER  |
|  266 | Canales      | Canales      | Sandra   | Conserje      | MUJER  |
|  476 | Perez        | Canales      | Abraham  | Jefe          | HOMBRE |
|   73 | Sanchez      | Casas        | Tulio    | Guardia       | HOMBRE |
|  164 | Paredes      | Herrera      | Sonia    | Empleado Gral | MUJER  |
|   37 | Casas        | Gutierrez    | Juan     | Encargad@     | HOMBRE |
|  245 | Herrera      | Huerta       | Ana      | Gerente       | MUJER  |
|   24 | Gutierrez    | Huerta       | Sonia    | Guardia       | MUJER  |
|  440 | Perez        | Villalobos   | Ana      | Conserje      | MUJER  |
|  155 | Chavez       | Paredes      | Vanessa  | Guardia       | MUJER  |
|  143 | Huerta       | Sanchez      | Carlos   | Empleado Gral | HOMBRE |
|  243 | Casas        | Herrera      | Vanessa  | Supervisor    | MUJER  |

In [ ]:
data = pd.DataFrame(listadatos)
data.to_csv('/content/drive/MyDrive/proyecto/salida.csv', index=False)

In [ ]:
data = pd.DataFrame(listadatos)
data.to_excel('/content/drive/MyDrive/proyecto/ejemplo.xlsx', index=False)

In [ ]:
path = "/content/drive/MyDrive/proyecto/datos.md"
data.to_markdown(path, index=False)

In [ ]:
path = "/content/drive/MyDrive/proyecto/salida.csv"

datos_csv = pd.read_csv(path)
print(datos_csv)

In [ ]:
path = "/content/drive/MyDrive/proyecto/holamundo.txt"

with open(path, "r") as f:
    contenido = f.read()

print(contenido)

  _____         _            __     __        _     
 |  __ \       | |           \ \   / /       (_)    
 | |__) |__  __| |_ __ ___    \ \_/ /__ _ __  _ ____
 |  ___/ _ \/ _` | '__/ _ \    \   / _ \ '_ \| |_  /
 | |  |  __/ (_| | | | (_) |    | |  __/ |_) | |/ / 
 |_|   \___|\__,_|_|  \___/     |_|\___| .__/|_/___|
                                       | |          
                                       |_|         
